File: src/types/session.ts
export type SessionTabKind = 'workspace' | 'file';

export interface SessionTab {
  id: string;
  kind: SessionTabKind;
  workspacePath: string;
  workspaceName: string;
  filePath: string | null;
  fileName: string | null;
  fileExtension: string | null;
  currentDirectory: string | null;
  position: number;
  active: boolean;
}

File: src-tauri/src/model/session.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
#[serde(rename_all = "camelCase")]
pub struct SessionTab {
    pub id: String,
    pub kind: String,
    pub workspace_path: String,
    pub workspace_name: String,
    pub file_path: Option<String>,
    pub file_name: Option<String>,
    pub file_extension: Option<String>,
    pub current_directory: Option<String>,
    pub position: i64,
    pub active: bool,
}
